src/lib/y2storage/proposal/devicegraph_generator.rb

```diff
@@ -87,6 +88,17 @@ def devicegraph(volumes, initial_graph, disk_analyzer)
 
     protected
 
+      # Modifies planned volumes adding an encryption password.
+      # Only volumes that can_live_on_logical_volume could have
+      # an encryption password.
+      # @param volumes [PlannedVolumesList]
+      def add_encryption_password(volumes)
```

Comment:
It is a bit rare to use bang method here due to it does not modify the state of the object. However, it would be consistent if we use that in other places. 

And about returning the list of volumes, I agree with you, in this case it is better to return whatever other thing.